build.yaml

```diff
@@ -1540,15 +1540,26 @@ steps:
       - default_ns
       - create_certs
   - kind: runImage
-    name: create_dummy_oauth2_client_secret
+    name: create_auth_config
```

Comment:
What's the benefit of this as opposed to inlining the secret into the deployment.yaml as an envvar?

Reply:
I wanted to allow `make -C auth deploy` to keep working without adding additional complexity to the Makefile. By putting it in the deployment, the Makefile would need to somehow know which oauth2 callback CI assigned that namespace. But this way you can make deploy into your dev namespace without worrying about the callback. I did this with `ci-config` as well.

More philosophically, I feel like so far things that are templated into the deployment.yaml are things that the Makefile (or whoever is doing the templating) owns like SHA, images, etc. the oauth2 callback is not one of those things since it is a property of the namespace and follows the namespace's lifecycle. In that way it's more akin to the global-config, but it felt easier to manage in a separate secret than stuffing it into the global-config. Happy to take suggestions on simplifying things here though.